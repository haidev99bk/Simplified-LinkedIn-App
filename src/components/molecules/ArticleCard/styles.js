import {StyleSheet} from 'react-native';
import color from '../../../styles/colors';
import fontWeight from '../../../styles/fontWeight';

const styles = StyleSheet.create({
  container: {
    backgroundColor: color.white,
    padding: 8,
    marginBottom: 8,
  },
  headerView: {
    flexDirection: 'row',
    width: '100%',
  },
  avatarContainer: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  userInfoContainer: {
    marginLeft: 4,
    flex: 1,
  },
  headerRight: {},
  contentView: {
    marginTop: 8,
  },
  imageAndTitleView: {
    backgroundColor: color.lightGrey,
    marginHorizontal: -8,
  },
  articleTitle: {
    padding: 8,
    ...fontWeight.bold,
  },
  thumbnailImage: {
    width: '100%',
    height: 220,
  },
});

export default styles;
