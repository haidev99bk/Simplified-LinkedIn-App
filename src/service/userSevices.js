const UserAndFollowers = require('./mockData/users.json');

export function getUserInfo() {
  return UserAndFollowers.users[0];
}
