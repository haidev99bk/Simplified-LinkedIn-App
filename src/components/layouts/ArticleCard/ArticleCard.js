import {View, Text, Image} from 'react-native';
import React, {useCallback} from 'react';
import styles from './styles';
import ArticleActionsBottomBar from '../../organisms/ArticleActionsBottomBar';
import ArticleHeaderBar from '../../organisms/ArticleHeaderBar';

const ArticleCard = props => {
  const {user, article, handleLike, hasLiked} = props;
  var dateAr = article.publicationDate.split('-');
  var newDate = dateAr[2] + '-' + dateAr[1] + '-' + dateAr[0].slice(-2);

  const handleLikeAction = useCallback(
    articleId => {
      if (handleLike) {
        handleLike(articleId);
      }
    },
    [handleLike],
  );

  const cardActions = [
    {
      iconFamily: 'AntDesign',
      iconName: 'like2',
      label: 'Like',
      onPress: handleLikeAction,
    },
    {
      iconFamily: 'Foundation',
      iconName: 'loop',
      label: 'Repost',
    },
    {
      iconFamily: 'FontAwesome',
      iconName: 'comment',
      label: 'Comment',
    },
    {
      iconFamily: 'Feather',
      iconName: 'send',
      label: 'Send',
    },
  ];

  const likesNumber = hasLiked ? article.likes + 1 : article.likes;

  return (
    <View style={styles.container}>
      <ArticleHeaderBar
        avatar={user.avatar}
        name={user.name}
        publishDate={newDate}
      />
      <View style={styles.contentView}>
        <Text style={styles.description}>{article.description}</Text>
        <View style={styles.imageAndTitleView}>
          <Image
            source={{uri: article.thumbnailImage}}
            style={styles.thumbnailImage}
          />
          <Text style={styles.articleTitle}>{article.title}</Text>
        </View>
      </View>
      <ArticleActionsBottomBar
        articleId={article.articleId}
        actions={cardActions}
        likes={likesNumber}
        comments={article.comments}
        hasLiked={hasLiked}
      />
    </View>
  );
};

export default React.memo(ArticleCard);
