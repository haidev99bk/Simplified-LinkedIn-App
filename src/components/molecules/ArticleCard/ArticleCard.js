import {View, Text, Image} from 'react-native';
import React from 'react';
import AvatarIcon from '../../atoms/AvatarIcon';
import styles from './styles';
import Icon from '../../atoms/Icon';

export default function ArticleCard(props) {
  const {user, article} = props;
  console.log('props ==> ', props);
  var dateAr = article.publicationDate.split('-');
  var newDate = dateAr[1] + '-' + dateAr[2] + '-' + dateAr[0].slice(-2);

  return (
    <View style={styles.container}>
      <View style={styles.headerView}>
        <View style={styles.avatarContainer}>
          <AvatarIcon source={{uri: user.avatar}} rounded={false} size={30} />
        </View>
        <View style={styles.userInfoContainer}>
          <Text>{user.name}</Text>
          <Text>{newDate}</Text>
        </View>
        <View style={styles.headerRight}>
          <Icon family="Entypo" name="dots-three-vertical" />
        </View>
      </View>
      <View style={styles.contentView}>
        <Text>{article.description}</Text>
        <View style={styles.imageAndTitleView}>
          <Image
            source={{uri: article.thumbnailImage}}
            style={styles.thumbnailImage}
          />
          <Text style={styles.articleTitle}>{article.title}</Text>
        </View>
      </View>
      <View style={styles.bottomView}>
        <View></View>
      </View>
    </View>
  );
}
