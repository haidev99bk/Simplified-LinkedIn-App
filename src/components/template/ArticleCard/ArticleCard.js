import {View, Text, Image} from 'react-native';
import React from 'react';
import styles from './styles';
import ArticleActionsBottomBar from '../../organisms/ArticleActionsBottomBar';
import ArticleHeaderBar from '../../organisms/ArticleHeaderBar';

export default function ArticleCard(props) {
  const {user, article} = props;
  var dateAr = article.publicationDate.split('-');
  var newDate = dateAr[2] + '-' + dateAr[1] + '-' + dateAr[0].slice(-2);

  const handleLikeAction = () => {};

  const cardActions = [
    {
      iconFamily: 'AntDesign',
      iconName: 'like2',
      label: 'Like',
      onPress: handleLikeAction,
    },
    {
      iconFamily: 'Foundation',
      iconName: 'loop',
      label: 'Repost',
    },
    {
      iconFamily: 'FontAwesome',
      iconName: 'comment',
      label: 'Comment',
    },
    {
      iconFamily: 'Feather',
      iconName: 'send',
      label: 'Send',
    },
  ];

  return (
    <View style={styles.container}>
      <ArticleHeaderBar
        avatar={user.avatar}
        name={user.name}
        publishDate={newDate}
      />
      <View style={styles.contentView}>
        <Text style={styles.description}>{article.description}</Text>
        <View style={styles.imageAndTitleView}>
          <Image
            source={{uri: article.thumbnailImage}}
            style={styles.thumbnailImage}
          />
          <Text style={styles.articleTitle}>{article.title}</Text>
        </View>
      </View>
      <ArticleActionsBottomBar
        actions={cardActions}
        likes={article.likes}
        comments={article.comments}
      />
    </View>
  );
}
