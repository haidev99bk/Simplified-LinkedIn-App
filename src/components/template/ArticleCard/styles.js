import {StyleSheet} from 'react-native';
import color from '../../../styles/colors';
import fontWeight from '../../../styles/fontWeight';
import {typography} from '../../../styles/typography';

const styles = StyleSheet.create({
  container: {
    backgroundColor: color.white,
    padding: 16,
    marginBottom: 16,
  },
  contentView: {
    marginTop: 8,
  },
  description: {
    ...typography.paragraph,
  },
  imageAndTitleView: {
    backgroundColor: color.lightGrey,
    marginHorizontal: -16,
    paddingBottom: 16,
    marginTop: 8,
  },
  articleTitle: {
    padding: 16,
    ...fontWeight.bold,
  },
  thumbnailImage: {
    width: '100%',
    height: 220,
  },
});

export default styles;
