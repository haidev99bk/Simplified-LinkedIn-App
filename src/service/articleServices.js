import {useId} from 'react';
import {getUserInfo} from './userSevices';
import AsyncStorage from '@react-native-async-storage/async-storage';

const Articles = require('./mockData/uArticles.json');

export function getArticles(userId) {
  const user = getUserInfo(userId);
  return Articles.articles.filter(a => {
    return user.articlesId.includes(a.articleId);
  });
}
