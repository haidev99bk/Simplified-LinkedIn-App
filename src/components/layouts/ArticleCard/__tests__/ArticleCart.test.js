import 'react-native';
jest.mock('react-native-device-info', () => {
  return {
    hasNotch: () => true,
  };
});
import React from 'react';
import renderer from 'react-test-renderer';
import {getArticles} from '../../../../service/articleServices';
import {getUserInfo} from '../../../../service/userSevices';
import ArticleCard from '../ArticleCard';

test('ArticleCard rendered correctly', () => {
  const snap = renderer
    .create(
      <ArticleCard
        user={getUserInfo(1)}
        article={getArticles(1)[0]}
        handleLike={() => {}}
        hasLiked={true}
      />,
    )
    .toJSON();

  expect(snap).toMatchSnapshot();
});
