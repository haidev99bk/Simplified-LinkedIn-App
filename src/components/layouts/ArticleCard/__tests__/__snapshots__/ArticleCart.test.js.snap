// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard rendered correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "marginBottom": 16,
      "padding": 16,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Image
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        source={
          {
            "uri": "https://randomuser.me/api/portraits/thumb/men/1.jpg",
          }
        }
        style={
          [
            {
              "height": 24,
              "width": 24,
            },
            {
              "height": 30,
              "width": 30,
            },
            false,
            undefined,
          ]
        }
      />
    </View>
    <View
      style={
        {
          "flex": 1,
          "marginLeft": 4,
        }
      }
    >
      <Text>
        Alice
      </Text>
      <Text>
        10-03-22
      </Text>
    </View>
    <View
      style={{}}
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": undefined,
              "fontSize": 12,
            },
            undefined,
            {
              "fontFamily": "Entypo",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "marginTop": 8,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 14,
          "lineHeight": 16,
        }
      }
    >
      In this article, we explore the numerous benefits of practicing yoga, including improved flexibility, stress relief, and mental clarity.
    </Text>
    <View
      style={
        {
          "backgroundColor": "#f8f9f9",
          "marginHorizontal": -16,
          "marginTop": 8,
          "paddingBottom": 16,
        }
      }
    >
      <Image
        source={
          {
            "uri": "https://picsum.photos/id/22/600/300",
          }
        }
        style={
          {
            "height": 220,
            "width": "100%",
          }
        }
      />
      <Text
        style={
          {
            "fontWeight": "700",
            "padding": 16,
          }
        }
      >
        The Benefits of Yoga for Mind and Body
      </Text>
    </View>
  </View>
  <View>
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingVertical": 8,
        }
      }
    >
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#1566c2",
              "borderRadius": 12,
              "height": 20,
              "justifyContent": "center",
              "marginRight": 8,
              "width": 20,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#fff",
                  "fontSize": 12,
                },
                undefined,
                {
                  "fontFamily": "anticon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text>
          16
        </Text>
      </View>
      <View>
        <Text>
          15 comments
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#f3f2ee",
          "height": 1,
        }
      }
    />
    <View>
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "padding": 4,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "height": 24,
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "height": 20,
                  "justifyContent": "center",
                  "width": 20,
                }
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "#1566c2",
                      "fontSize": 16,
                    },
                    undefined,
                    {
                      "fontFamily": "anticon",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                
              </Text>
            </View>
            <Text
              style={
                {
                  "fontWeight": "600",
                }
              }
            >
              Like
            </Text>
          </View>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "height": 24,
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "height": 20,
                  "justifyContent": "center",
                  "width": 20,
                }
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": undefined,
                      "fontSize": 16,
                    },
                    undefined,
                    {
                      "fontFamily": "fontcustom",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                
              </Text>
            </View>
            <Text
              style={
                {
                  "fontWeight": "600",
                }
              }
            >
              Repost
            </Text>
          </View>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "height": 24,
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "height": 20,
                  "justifyContent": "center",
                  "width": 20,
                }
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": undefined,
                      "fontSize": 16,
                    },
                    undefined,
                    {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                
              </Text>
            </View>
            <Text
              style={
                {
                  "fontWeight": "600",
                }
              }
            >
              Comment
            </Text>
          </View>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "height": 24,
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "height": 20,
                  "justifyContent": "center",
                  "width": 20,
                }
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": undefined,
                      "fontSize": 16,
                    },
                    undefined,
                    {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                
              </Text>
            </View>
            <Text
              style={
                {
                  "fontWeight": "600",
                }
              }
            >
              Send
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
